src/json-patch-queue-synchronous.js

```diff
@@ -114,7 +121,7 @@ JSONPatchQueueSynchronous.prototype.reset = function(obj, newState){
 	this.version = JSONPatchQueueSynchronous.getPropertyByJsonPointer(newState, this.versionPath);
 	this.waiting = [];
 	var patch = [{ op: "replace", path: "", value: newState }];
-	this.apply(obj, patch);
+	return this.obj = this.apply(obj, patch);
```

Comment:
I think that puts another requirement for `apply` callback to return an updated object. The requirement that needs to be documented. (in JSDoc and README)